File: programmers/js_src/level01/76501.js
// 음양 더하기

function solution(absolutes, signs) {
  var answer = 0;

  answer = absolutes
    .map((num, index) => !signs[index] ? -num : num)
    .reduce((sum, cur) => sum + cur, 0);

  return answer;
}

// 다른 풀이
// function solution(absolutes, signs) {
//   return absolutes.reduce((acc, val, i) => acc + (val * (signs[i] ? 1 : -1)), 0);
// }

File: programmers/java/level01/76501.java
// 음양 더하기

import java.util.stream.IntStream;

class Solution {
    public int solution(int[] absolutes, boolean[] signs) {
        return IntStream.range(0, absolutes.length)
            .map(idx -> signs[idx] ? absolutes[idx] : -absolutes[idx])
            .sum();
    }
}